src/osdc/Objecter.h

```diff
@@ -2743,6 +2745,11 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   void dump_pool_stat_ops(ceph::Formatter *fmt) const;
   void dump_statfs_ops(ceph::Formatter *fmt) const;
 
+  int get_num_op_laggy() const { return num_op_laggy; }
+  void set_num_op_laggy(int num) { num_op_laggy = num; } 
+  int get_num_osd_laggy() const { return num_osd_laggy; }
+  void set_num_osd_laggy(int num) { num_osd_laggy = num; } 
+
```

Comment:
maybe you can just use `{ return logger->get(l_osdc_osd_laggy)}` instead of adding new `num_osd_laggy` ?

Reply:
yeah, didn't read perf_counters code, this can be used. will push the change.

Reply:
I cannot do this in `Objecter.h` as `l_osdc_osd_laggy` is part of enum in `Objecter.cc`. I need to define getter functions in `Objecter.cc`, would it be a good approach to expose these functions there?

Reply:
Is there a way to get the perf counters directly instead of adding a new helper ?

Reply:
@lxbsz there are 66 elements in the enum with it's first element initialized `l_osdc_first = 123200` and `l_osdc_osd_laggy` is [62th](https://github.com/ceph/ceph/blob/main/src/osdc/Objecter.cc#L176) so it should be `123262` and looking at `void PerfCounters::set(int idx, uint64_t amt)`, i think it would initialize `123262` with the value of laggy_osds right?

Reply:
Yeah, it is. BTW, why do you need to compute this ? You can just use the `l_osdc_osd_laggy` instead.

Reply:
if we do it in `.cc` then yes, but if we wan to do it `.h` then need to access it using its `int` value, also `mds->objecter->logger->get(123262)` is not possible in `Server.cc` directly as `logger` is private in Objecter class

Reply:
`mds->objecter->logger->get(123262)` this looks incorrect.

Why not just adding a public helper in `Objecter` class ?

Reply:
> `mds->objecter->logger->get(123262)` this looks incorrect.

by this i meant direct usage in `Server.cc`



> Why not just adding a public helper in `Objecter` class ?


Right, so if you want to place it in .cc then we can use `l_osdc_osd_laggy` directly but in `.h` we need to use int value as enum values from `.cc` wont be accessible in `.h`